Problem with F() objects

I have model:

```
class Versioned(models.Model):
    version = models.IntegerField(editable=False)

    def save(self, *args, **kwargs):
        self.version = 1 if not self.version else F('version') + 1
        super().save(*args, **kwargs)
```

And this cause a problem on `save()` call:

```

    def test_increment_version(factory):
        obj = factory.create()
        version1 = obj.version
>       obj.save()

contacts/tests/test_models.py:53: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
core/models.py:92: in save
    super().save(*args, **kwargs)
../../.virtualenvs/piksoftware/lib/python3.6/site-packages/django/db/models/base.py:808: in save
    force_update=force_update, update_fields=update_fields)
../../.virtualenvs/piksoftware/lib/python3.6/site-packages/django/db/models/base.py:848: in save_base
    update_fields=update_fields, raw=raw, using=using,
../../.virtualenvs/piksoftware/lib/python3.6/site-packages/django/dispatch/dispatcher.py:193: in send
    for receiver in self._live_receivers(sender)
../../.virtualenvs/piksoftware/lib/python3.6/site-packages/django/dispatch/dispatcher.py:193: in <listcomp>
    for receiver in self._live_receivers(sender)
../../.virtualenvs/piksoftware/lib/python3.6/site-packages/simple_history/models.py:267: in post_save
    self.create_historical_record(instance, created and '+' or '~')
../../.virtualenvs/piksoftware/lib/python3.6/site-packages/simple_history/models.py:282: in create_historical_record
    history_change_reason=history_change_reason, **attrs)
../../.virtualenvs/piksoftware/lib/python3.6/site-packages/django/db/models/manager.py:85: in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
../../.virtualenvs/piksoftware/lib/python3.6/site-packages/django/db/models/query.py:394: in create
    obj.save(force_insert=True, using=self.db)
../../.virtualenvs/piksoftware/lib/python3.6/site-packages/django/db/models/base.py:808: in save
    force_update=force_update, update_fields=update_fields)
../../.virtualenvs/piksoftware/lib/python3.6/site-packages/django/db/models/base.py:838: in save_base
    updated = self._save_table(raw, cls, force_insert, force_update, using, update_fields)
../../.virtualenvs/piksoftware/lib/python3.6/site-packages/django/db/models/base.py:924: in _save_table
    result = self._do_insert(cls._base_manager, using, fields, update_pk, raw)
../../.virtualenvs/piksoftware/lib/python3.6/site-packages/django/db/models/base.py:963: in _do_insert
    using=using, raw=raw)
../../.virtualenvs/piksoftware/lib/python3.6/site-packages/django/db/models/manager.py:85: in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
../../.virtualenvs/piksoftware/lib/python3.6/site-packages/django/db/models/query.py:1076: in _insert
    return query.get_compiler(using=using).execute_sql(return_id)
../../.virtualenvs/piksoftware/lib/python3.6/site-packages/django/db/models/sql/compiler.py:1112: in execute_sql
    for sql, params in self.as_sql():
../../.virtualenvs/piksoftware/lib/python3.6/site-packages/django/db/models/sql/compiler.py:1065: in as_sql
    for obj in self.query.objs
../../.virtualenvs/piksoftware/lib/python3.6/site-packages/django/db/models/sql/compiler.py:1065: in <listcomp>
    for obj in self.query.objs
../../.virtualenvs/piksoftware/lib/python3.6/site-packages/django/db/models/sql/compiler.py:1064: in <listcomp>
    [self.prepare_value(field, self.pre_save_val(field, obj)) for field in fields]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <django.db.models.sql.compiler.SQLInsertCompiler object at 0x107051cc0>, field = <django.db.models.fields.IntegerField: version>
value = <CombinedExpression: Col(contacts_historicalcomment, contacts.HistoricalComment.version) + Value(1)>

    def prepare_value(self, field, value):
        """
            Prepare a value to be used in a query by resolving it if it is an
            expression and otherwise calling the field's get_db_prep_save().
            """
        if hasattr(value, 'resolve_expression'):
            value = value.resolve_expression(self.query, allow_joins=False, for_save=True)
            # Don't allow values containing Col expressions. They refer to
            # existing columns on a row, but in the case of insert the row
            # doesn't exist yet.
            if value.contains_column_references:
                raise ValueError(
                    'Failed to insert expression "%s" on %s. F() expressions '
>                   'can only be used to update, not to insert.' % (value, field)
                )
E               ValueError: Failed to insert expression "Col(contacts_historicalcomment, contacts.HistoricalComment.version) + Value(1)" on contacts.HistoricalComment.version. F() expressions can only be used to update, not to insert.

../../.virtualenvs/piksoftware/lib/python3.6/site-packages/django/db/models/sql/compiler.py:999: ValueError
```

It looks like a "by design" problem. But I think we should write something in docs about it.
We can use `refresh_from_db()`, but the result of this call may return a quite different version than we are saved. 

Any IDEAs?

--- a/docs/reference.rst
+++ b/docs/reference.rst
@@ -39,3 +39,14 @@ you use:
     from simple_history.middleware import HistoricalRecords
     if hasattr(HistoricalRecords.thread, 'request'):
         del HistoricalRecords.thread.request
+
+Using F() expressions
+---------------------
+``F()`` expressions, as described here_, do not work on models that have
+history. Simple history inserts a new record in the historical table for any
+model being updated. However, ``F()`` expressions are only functional on updates.
+Thus, when an ``F()`` expression is used on a model with a history table, the
+historical model tries to insert using the ``F()`` expression, and raises a
+``ValueError``.
+
+.. _here: https://docs.djangoproject.com/en/2.0/ref/models/expressions/#f-expressions

